app/src/main/java/org/mozilla/fenix/components/metrics/AppAllSourceStartTelemetry.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.content.Intent
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ProcessLifecycleOwner
+import mozilla.components.support.utils.SafeIntent
+
+/**
+ * Tracks how the application was opened through [Event.AppOpenedAllSourceStartup].
+ * We only considered to be "opened" if it received an intent and the app was in the background.
+ */
+class AppAllSourceStartTelemetry(private val metrics: MetricController) : LifecycleObserver {
+
+    // default value is true to capture the first launch of the application
+    private var wasApplicationInBackground = true
+
+    init {
+        ProcessLifecycleOwner.get().lifecycle.addObserver(this)
```

Comment:
No action needed: since this is lazily loaded and we only attach the `ProcessLifecycleOwner` in the initializer, it's possible we may miss `onStop` if the code is refactored such that we don't get called in `*Activity.onCreate`. Ideally, we'd separate the initialization from the first call to help avoid this but let's skip it because I don't think it's worth the time or additional churn.